components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3307,6 +3307,638 @@ func TestUpdatePolicy(t *testing.T) {
+				"chef-server-1", storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			condition2, err := storage.NewCondition(ruleType,
+				"org1,org2,org3", storage.Organization, storage.MemberOf)
+			require.NoError(t, err)
+			condition3, err := storage.NewCondition(ruleType,
+				"role1", storage.ChefRole, storage.Equals)
+			require.NoError(t, err)
+
+			rule, err := storage.NewRule("new-id-1", "project-1", "name", ruleType,
+				[]storage.Condition{condition1, condition2, condition3})
+			require.NoError(t, err)
+			resp, err := store.CreateRule(ctx, &rule)
+			require.NoError(t, err)
+			require.Equal(t, &rule, resp)
+		},
+		"create event rule with multiple conditions": func(t *testing.T) {
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+
+			ruleType := storage.Node
+			condition1, err := storage.NewCondition(ruleType,
+				"chef-server-1", storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			condition2, err := storage.NewCondition(ruleType,
+				"org1,org2,org3", storage.Organization, storage.MemberOf)
+			require.NoError(t, err)
+			condition3, err := storage.NewCondition(ruleType,
+				"chef-server-2,chef-server-3", storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+
+			rule, err := storage.NewRule("new-id-1", "project-1", "name", ruleType,
+				[]storage.Condition{condition1, condition2, condition3})
+			require.NoError(t, err)
+			resp, err := store.CreateRule(ctx, &rule)
+			require.NoError(t, err)
+			require.Equal(t, &rule, resp)
+		},
+	}
+
+	for name, test := range cases {
+		t.Run(name, test)
+		db.flush(t)
+	}
+}
+
+func TestListRules(t *testing.T) {
+	store, db, _ := setup(t)
+	defer db.close(t)
+	defer store.Close()
+
+	cases := map[string]func(*testing.T){
+		"when no rules exist, returns an empty list": func(t *testing.T) {
+			ctx := context.Background()
+			resp, err := store.ListRules(ctx)
+			assert.NoError(t, err)
+			assert.Nil(t, resp)
+			assert.Equal(t, 0, len(resp))
+		},
+		"when mutiple rules exist with no project fitler, returns the full list": func(t *testing.T) {
+			ctx := context.Background()
+
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+
+			ruleType := storage.Node
+			condition1, err := storage.NewCondition(ruleType,
+				"chef-server-1", storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			condition2, err := storage.NewCondition(ruleType,
+				"org1,org2,org3", storage.Organization, storage.MemberOf)
+			require.NoError(t, err)
+			condition3, err := storage.NewCondition(ruleType,
+				"chef-server-2,chef-server-3", storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule1, err := storage.NewRule("new-id-1", projID, "name", ruleType,
+				[]storage.Condition{condition1, condition2, condition3})
+			require.NoError(t, err)
+			_, err = store.CreateRule(ctx, &rule1)
```

Comment:
⚠️ The test for `ListRules` now depends on `CreateRule`. Let's try to avoid that and do a straightforward DB query to set the stage database-wise for these tests.

Reply:
That adds so much complexity imo. I'm really not a big fan of all of our tests that do that. So much to refactor if you change anything.

Reply:
Mostly I'm not following the logic here. We have tests for CreateRule and UpdateRule. We are confident they work. If they don't those tests will break. If we aren't confident those work, we should expand those tests?

Reply:
If we were changing, say, the database layout in some of way, the only approach would be all-at-once. The different CRUD methods could not be worked on in increments or in parallel.

It's always a tradeoff, but this is a learning from previous difficult situations we had gotten  ourselves into. 😉

Reply:
After chatting with the team, I added precondition assertions so we felt confident that the calls to `CreateRule` weren't giving us false positives elsewhere. If you still feel super strongly, would it be ok if we merged as is and refactored so we aren't blocking the rest of the story?